sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -497,7 +497,11 @@ object DataSourceStrategy {
         Some(sources.IsNotNull(a.name))
 
       case expressions.And(left, right) =>
-        (translateFilter(left) ++ translateFilter(right)).reduceOption(sources.And)
+        // See SPARK-12218 and PR 10362 for detailed discussion
```

Comment:
In the comment, you need to give an example to explain why.

Reply:
Sure. I have added more comments there with an example. Thanks, Sean!

Reply:
Usually we don't list PR number but just JIRA number is enough.

Reply:
@viirya I see. Thanks, Simon! I've removed the PR number from the comment.